src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -675,16 +673,30 @@ static void AddPropertyToTypeDeclaration(CodeTypeDeclaration typeDeclaration, Pr
                 ? new CodeTypeReference(member.PropertyType.Name)
                 : member.PropertyType.CreateCodeTypeReference(member);
 
+            var propertyName = member.Name;
             var property = new CodeMemberProperty
             {
-                Name = member.Name,
+                Name = propertyName,
                 Type = propertyType,
                 Attributes = propertyAttributes,
                 CustomAttributes = CreateCustomAttributes(member, attributeFilter),
                 HasGet = hasGet,
                 HasSet = hasSet
             };
 
+            // DefaultMemberAttribute on type gets propagated to IndexerNameAttribute
+            var defaultMemberAttribute = typeDeclarationInfo.CustomAttributes.SingleOrDefault(x =>
+                x.AttributeType.FullName == "System.Reflection.DefaultMemberAttribute");
+            var defaultMemberAttributeArgument = defaultMemberAttribute?.ConstructorArguments.SingleOrDefault();
+            var value = defaultMemberAttributeArgument?.Value as string;
+            if (!string.IsNullOrEmpty(value) && propertyName != "Item")
+            {
+                property.Name = "Item";
```

Comment:
What would happen without this line?

Reply:
I'm guessing this is the line that makes CodeDOM stop rendering it as a non-indexer property.

Reply:
Correct. It requires Item